sw/ip/mbx/dif/dif_mbx.c

```diff
@@ -15,16 +15,18 @@ dif_result_t dif_mbx_range_set(const dif_mbx_t *mbx,
   if (mbx == NULL) {
     return kDifBadArg;
   }
-  if (config.imbx_base_addr >= config.imbx_limit_addr) {
+  // Note: the limit addresses are _inclusive_, specifying the start address of
+  // the final valid DWORD.
+  if (config.imbx_base_addr > config.imbx_limit_addr) {
     return kDifBadArg;
   }
-  if (config.ombx_base_addr >= config.ombx_limit_addr) {
+  if (config.ombx_base_addr > config.ombx_limit_addr) {
     return kDifBadArg;
   }
-  // Check that the inbound mailbox and outbound mailbox memory ranges collide
-  // with each other.
-  if ((config.imbx_base_addr < config.ombx_limit_addr) &&
-      (config.ombx_base_addr < config.imbx_limit_addr)) {
+  // Check that the inbound mailbox and outbound mailbox memory ranges do not
+  // collide with each other.
+  if ((config.imbx_base_addr <= config.ombx_limit_addr) &&
+      (config.ombx_base_addr <= config.imbx_limit_addr)) {
```

Comment:
I agree with this change. It causes a CI test to fail, though, and I think that's because this unittest
https://github.com/lowRISC/opentitan/blob/ee9e6eaa1824418f26a984d6910229a7c53e2623/sw/ip/mbx/dif/dif_mbx_unittest.cc#L62-L63
needs to be changed so that inbound and outbound mailbox don't overlap.